### DISCUSSION TITLE
encryption files must live in chezmoi root but the tutorial doesn't mention it

### DISCUSSION THREAD

**Discussion by danielo515** [twpayne/chezmoi#2946] [General]
Again you for creating chezmoi. I recently followed the encryption tutorial, but it didn't mention that you must put things in the chezmoi root directory (in case is not the default). Could you please either mention this in the tutorial or make `chezmoi cd` take into account the right directory? Thank you for your time!

**Comment by twpayne** [comment]
I don't understand what you mean. Please could you open a PR with the changes to the documentation that you would like to see.

**Reply by danielo515** [reply]
https://github.com/twpayne/chezmoi/pull/2948

**Comment by halostatue** [comment]
@danielo515 Do you mean that `key.txt` (or whatever) needs to be in the path specified by `.chezmoiroot`? That’s somewhat expected, as `.chezmoiscripts` and the other files important to Chezmoi need to be there and not in the working tree.

**Reply by danielo515** [reply]
I'm talking about this docs: https://www.chezmoi.io/user-guide/frequently-asked-questions/encryption/
If you follow that step by step tutorial you will end with things in the wrong place because `chezmoi cd` does not put you in the chezmoi root directory, it puts you in the repository root, which in some cases it is the same as chezmoi root, and in some cases it is not.
I think docs should make easy for all users to follow them, and here making a note about the fact that you may need to get into a different folder is definitely something I will be grateful to know beforehand.

**Reply by halostatue** [reply]
`chezmoi cd ~` will put you in `.chezmoiroot`, rather than the working tree (the "source" directory), if they are different.

While I agree with your general sentiment (improved documentation for all users to follow), chezmoi is feature-rich enough with enough features that *interact* with one another that the resulting documentation would be much *harder* to follow. You have found this particular sharp edge with encryption, but there are a half-dozen other features which are affected by the *explicit* use of `.chezmoiroot` or some other feature interaction. I will be making suggestions for you on #2948 that will result in, IMO, better documentation with less repetition.
